monitoring/ping_test.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+	"net"
+
+	"github.com/gravitational/satellite/agent"
+	"github.com/gravitational/satellite/agent/health"
+	"github.com/gravitational/satellite/agent/proto/agentpb"
+
+	serf "github.com/hashicorp/serf/client"
+	"github.com/hashicorp/serf/coordinate"
+	"gopkg.in/check.v1"
+)
+
+type PingSuite struct{}
+
+var _ = check.Suite(&PingSuite{})
+
+func (*PingSuite) TestPingChecker(c *check.C) {
+	testCases := []struct {
+		Nodes       []serf.Member
+		Coords      map[string]*coordinate.Coordinate
+		Status      agentpb.NodeStatus_Type
+		Description string
+	}{
+		{
+			Nodes: []serf.Member{
+				{
+					Name:   "member-1",
+					Status: agentpb.MemberStatus_Alive.String(),
+					Addr:   net.IPv4(127, 0, 0, 1),
+				},
+				{
+					Name:   "member-2",
+					Status: agentpb.MemberStatus_Alive.String(),
+					Addr:   net.IPv4(127, 0, 0, 2),
+				},
+			},
+			Coords: map[string]*coordinate.Coordinate{
+				"member-1": &coordinate.Coordinate{
+					Height: 0.001,
+				},
+				"member-2": &coordinate.Coordinate{
+					Height: 0.001,
+				},
+			},
+			Status:      agentpb.NodeStatus_Running,
+			Description: "Testing standard working condition with values below threshold",
+		},
+		{
+			Nodes: []serf.Member{
+				{
+					Name:   "member-1",
+					Status: agentpb.MemberStatus_Alive.String(),
+					Addr:   net.IPv4(127, 0, 0, 1),
+				},
+				{
+					Name:   "member-2",
+					Status: agentpb.MemberStatus_Alive.String(),
+					Addr:   net.IPv4(127, 0, 0, 2),
+				},
+			},
+			Coords:      map[string]*coordinate.Coordinate{},
+			Status:      agentpb.NodeStatus_Degraded,
+			Description: "Testing missing coordinates for cluster members",
+		},
+		{
+			Nodes: []serf.Member{
+				{
+					Name:   "member-1",
+					Status: agentpb.MemberStatus_Alive.String(),
+					Addr:   net.IPv4(127, 0, 0, 1),
+				},
+				{
+					Name:   "member-2",
+					Status: agentpb.MemberStatus_Failed.String(),
+					Addr:   net.IPv4(127, 0, 0, 2),
+				},
+			},
+			Coords: map[string]*coordinate.Coordinate{
+				"member-1": &coordinate.Coordinate{
+					Height: 3600,
+				},
+				"member-2": &coordinate.Coordinate{
+					Height: 3600,
+				},
+			},
+			Status:      agentpb.NodeStatus_Running,
+			Description: "Testing nodes with latency value over the threshold",
```

Comment:
If the value is over the threshold - why is the probe running? The comment on `checkNodesRTT` explicitly states the check fails if the values are over the threshold - either the test should verify that the probe's status is `NodeStatus_Failed` or the comment rephrased to hint that this is merely a warning.

Reply:
I changed the comment to specify that it just raises a warning, instead of failing.

Reply:
I'm just wondering why this is a warning in a log instead of a failing checker? It will hardly draw attention in this case.

Reply:
I initially set it that way, but @r0mant suggested this to be a warning instead.

I'll change it back to fail.

Reply:
No, I think it is correct at this point. This is,in fact, more like a warning which we don't model in satellite just yet (on one hand it should be visible to the user, on the other it should not be terminal to the cluster state - i.e. should not report it as `degraded`).
We do have the [Warning](https://github.com/gravitational/satellite/blob/master/agent/proto/agentpb/agent.proto#L103) severity which I would try to use in this case (i.e. `probe.State=pb.Probe_Running` and `probe.Severity = pb.Probe_Warning`. This could be handled on the gravity side by additionally outputting probes with `severity == pb.Probe_Warning`.

Reply:
Added that. Please confirm it looks good and I'll (SQUASH &) merge everything. Thanks.